.changeset/fix-characterCounterProp.md

```diff
@@ -0,0 +1,7 @@
+---
+'react-magma-dom': patch
+---
+
+fix(Character Counter): Added two new props 'maxCount' and 'hasCharacterCounter'. The prop of 'maxCount' replaces 'maxLength'. Now 'maxLength' works as the native HTML attribute.
+
+Note: This update is a breaking change and all uses of the Character Counter need to update to the 'maxCount' prop name.
```

Comment:
The way you've built this, it is _not_ a breaking change (on purpose). I think it would be clearer to say something along these lines:

fix(CharacterCounter): 'maxLength' prop now works as the native HTML attribute if new `hasCharacterCounter` is set to `false`. Character Counter functionality is now tied to new `maxCount` prop. `hasCharacterCounter` is a temporary prop until our next major version where the use of `maxLength` for character counter will be deprecated.